lib/boss_rds.py

```diff
@@ -0,0 +1,213 @@
+# Copyright 2019 The Johns Hopkins University Applied Physics Laboratory
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from contextlib import contextmanager
+from collections import namedtuple
+from mysql import connector
+import logging
+
+from lib.external import ExternalCalls
+from lib.names import AWSNames
+from lib import aws
+
+@contextmanager
+def connect_rds(session, domain):
+    """
+    Context manager with established connection to rds
+
+    Args:
+        session (Session): Open boto3 session.
+        domain (str): VPC such as integration.boss.
+
+    Returns:
+        sql: connection context
+    """
+    mysql_params = get_mysql_params(session, domain)
+
+    keypair = aws.keypair_lookup(session)
+    call = ExternalCalls(session, keypair, domain)
+
+    logging.info('Tunneling to DB (slow) . . .')
+    with call.tunnel(mysql_params.host, mysql_params.port, 'rds') as local_port:
+        try:
+            sql = connector.connect(
+                user=mysql_params.user, password=mysql_params.password, 
+                port=local_port, database=mysql_params.db
+            )
+            yield sql
```

Comment:
Does it make more sense to return a cursor instead of the connection object? Currently you always create a cursor and have to remember to close it.

Reply:
Yeah I could do this, but then it wouldn't be general to sql right. It would have to include the cursor.... All the functionality I can see includes using the cursor, but maybe there is something else out there?

Reply:
True. Just thinking out loud.

Is it required to close the cursor before the closing the connection? I'm just wondering if someone doesn't use a `try / finally` to make sure it is closed.